cli/compose/template/template.go

```diff
@@ -7,7 +7,7 @@ import (
 )
 
 var delimiter = "\\$"
-var substitution = "[_a-z][_a-z0-9]*(?::?[-?][^}]*)?"
```

Comment:
It is only said that `do not begin with a digit`, so starting with `.` is valid? 🤔 
So the regexp could be: `[_a-z.][_a-z0-9.]*(?::?[-?][^}]*)?` ?

And by the way a small nit:
```go
var(
    delimiter = 
    substitution = 
    patternString =
    pattern =
)
```